### PR TITLE
[#499] autogenerated colors are checked for min brightness and saturation

### LINKED ISSUE #499: Contrast of autogenerated group colors too low
Depending on the background underneath the SVG autogenerated colors can sometimes be too light or too dark.

* allow only colors within a certain range 
* investigate HSV or HSL colour space

### PR DESCRIPTION
**Pull request by bonndan**
closes #499 

To play around with HSV colors, google a hex color: https://www.google.com/search?q=%2343FDC1

### CODE REVIEW

**Review by DanielBruenker — COMMENTED** [review]
Looks good overall!

File: src/main/java/de/bonndan/nivio/output/Color.java [hunk 1]
```diff
@@ -1,61 +1,113 @@
...
      */
-    public static String nameToRGB(String name) {
-        return nameToRGB(name, "FFFFFF");
+    @Nullable
+    public static String nameToRGB(@Nullable final String name, @Nullable final String defaultColor) {
+        if (StringUtils.isEmpty(name)) {
+            return defaultColor;
+        }
+
+        String colorString = nameToRGBRaw(name);
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/Color.java:37]: Hihi, yes that comment is misleading. I wanted it to be private, but in this particular case I also need the code in the tests to prepare the test conditions.

File: src/main/java/de/bonndan/nivio/output/Color.java [hunk 2]
```diff
@@ -1,61 +1,113 @@
...
+            return saturate(hsv[0], hsv[1], hsv[2]);
+        }
+
+        return colorString;
     }
 
-    public static String nameToRGB(String name, String defaultColor) {
-        if (StringUtils.isEmpty(name))
-            return defaultColor;
+    static float[] hsv(String hexColor) {
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/output/Color.java]: Fixed

File: src/main/java/de/bonndan/nivio/output/Color.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielBruenker) [reply src/main/java/de/bonndan/nivio/output/Color.java:37]: I see. Thank you for the explanation.

**Review by DanielBruenker — APPROVED** [review]

**Pull request merged**